jekyll/_cci2/rerun-failed-tests.adoc

```diff
@@ -270,12 +270,52 @@ steps:
 
-NOTE: The `halt` command will execute the rest of the _current_ step, regardless of whether `tests.txt` has content or not. Make sure to place the command to execute tests in the _following_ step. 
+NOTE: The `halt` command will execute the rest of the _current_ step, regardless of whether `tests.txt` has content or not. Make sure to place the command to execute tests in the _following_ step.
 
 * Orbs that run tests may not work with this new functionality at this time.
 * If a shell script is invoked to run tests, `circleci tests run` should be placed **in the shell script** itself, and not `.circleci/config.yml`. Alternatively, see the <<output-test-files-only,section above>> to pipe the list of test files to be run to a `.txt` file and then pass the list of test file names to your shell script.
 * Jobs that are older than the xref:persist-data#custom-storage-usage[retention period] for workspaces for the organization cannot be rerun with "Rerun failed tests".
-* Jobs that upload code coverage reports link:https://discuss.circleci.com/t/product-launch-re-run-failed-tests-only-circleci-tests-run/47775/3?u=sebastian-lerner[may see issues during a rerun].
+* Jobs that upload code coverage reports:
+
+To ensure that code coverage reports from the original job run are persisted to an artifact in addition to the report that is generated on a re-run, see the following example for a sample Go project:
+
+{% raw %}
+[source,yaml]
+----
+  - run:
+          name: Run tests with coverage being saved
+          command: go list ./... | circleci tests run --timings-type "name" --command "xargs gotestsum --junitfile junit.xml --format testname -- -coverprofile=cover.out"
+
+      # For a rerun that succeeds, restore the coverage files from the failed run
+      - restore_cache:
+          key: coverage-{{.Revision}}
+
+      # For a failed run, save the coverage for rerunning failed tests. CircleCI will skip saving if this revision key has already been saved.
+      - save_cache:
+          key: coverage-{{.Revision}}
+          paths:
+            - cover.out
+          when: on_fail
+
+      # Needed in order to enable rerun failed tests
+      - store_test_results:
+          path: junit.xml
+          when: always
+
+      # Upload coverage file html which includes all of the tests (not just the rerun)
+      - run:
+          name: Save html coverage
+          command: go tool cover -html=cover.out -o cover.html
+          when: always
+      - store_artifacts:
+          path: cover.html
+          when: always
+----
+{% endraw %}
+
+The snippet above uses the built-in xref:caching[`.Revision`] key to store a coverage report for the current vcs revision.  On a successful rerun, the original job run's coverage report will be restored to include the coverage from the skipped (passed) tests. It can then be used in a downstream job for aggregation or analysis.
```

Comment:
```suggestion
The snippet above uses the built-in xref:caching#[`.Revision`] key to store a coverage report for the current VCS revision.  On a successful rerun, the original job run's coverage report will be restored to include the coverage from the skipped (passed) tests. It can then be used in a downstream job for aggregation or analysis.
```